rest-server/core/src/main/scala/com/actionml/core/template/Dataset.scala

```diff
@@ -22,9 +22,9 @@ import com.actionml.core.storage.Store
 import com.actionml.core.validate.ValidateError
 import com.typesafe.scalalogging.LazyLogging
 
-abstract class Dataset[T](r: String) extends LazyLogging {
+abstract class Dataset[T](engineId: String) extends LazyLogging {
 
-  val resourceId: String = r
+  val resourceId: String = engineId
```

Comment:
this engineId alias looks confusing for me

Reply:
In the beginning Events and Queries were tied to Datasets and Engines. But experience showed an Engine is needed in all cases as a controller of the Dataset, so what was called a resourceId "r" is now always an engineId and so this should be less confusing I hope.